core/messaging/__init__.py

```diff
@@ -32,8 +38,35 @@ def __getitem__(self, item):
         return getattr(self, item)
 
     @property
-    def cell_indexes(self) -> str:
-        return ", ".join(self.__cell_indexes)
+    def cell_indexes(self) -> tuple[str, ...] | None:
```

Comment:
Now that `__cell_indexes` is the same as `cell_indexes`, maybe simpler just to have `cell_indexes` as a normal attribute with no `property` method?

Reply:
I'm mildly of two minds on removing the property. Having it as a 'protected' property means that it can't (easily) be set to an invalid value (see below) - forcing to either instantiate a new Message, which we know calls the `clean` method, or via combining messages, which also calls `clean`. I think it's a defensive design - but if you'd rather remove it and leave it up to the developer to do the right thing, I could probably be persuaded.

```python
>>> m.cell_indexes = 'foo'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: property 'cell_indexes' of 'Message' object has no setter
```

Reply:
Makes sense - gives us a firm guarantee that cell indexes will be correctly formatted and sorted - happy to leave as is and resolve then